src/append/rolling_file/policy/compound/trigger/time.rs

```diff
@@ -0,0 +1,614 @@
+//! The time trigger.
+//!
+//! Requires the `time_trigger` feature.
+
+#[cfg(test)]
+use chrono::NaiveDateTime;
+use chrono::{DateTime, Datelike, Duration, Local, TimeZone, Timelike};
+#[cfg(test)]
+use mock_instant::{SystemTime, UNIX_EPOCH};
+#[cfg(feature = "config_parsing")]
+use serde::de;
+#[cfg(feature = "config_parsing")]
+use std::fmt;
+use std::sync::RwLock;
+
+use crate::append::rolling_file::{policy::compound::trigger::Trigger, LogFile};
+
+#[cfg(feature = "config_parsing")]
+use crate::config::{Deserialize, Deserializers};
+
+#[cfg(feature = "config_parsing")]
+/// Configuration for the time trigger.
+#[derive(Copy, Clone, Eq, PartialEq, Hash, Debug, Default, serde::Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct TimeTriggerConfig {
+    interval: TimeTriggerInterval,
+    #[serde(default)]
+    modulate: bool,
+    #[serde(default)]
+    max_random_delay: u64,
+}
+
+#[cfg(not(feature = "config_parsing"))]
+/// Configuration for the time trigger.
+#[derive(Copy, Clone, Eq, PartialEq, Hash, Debug, Default)]
+pub struct TimeTriggerConfig {
+    interval: TimeTriggerInterval,
+    modulate: bool,
+    max_random_delay: u64,
+}
+
+/// A trigger which rolls the log once it has passed a certain time.
+#[derive(Debug)]
+pub struct TimeTrigger {
+    config: TimeTriggerConfig,
+    next_roll_time: RwLock<DateTime<Local>>,
+}
+
+/// The TimeTriger have the following units are supported (case insensitive):
+/// "second", "seconds", "minute", "minutes", "hour", "hours", "day", "days", "week", "weeks", "month", "months", "year", "years". The unit defaults to
+/// second if not specified.
+#[derive(Copy, Clone, Eq, PartialEq, Hash, Debug)]
+pub enum TimeTriggerInterval {
+    /// TimeTriger in second(s).
+    Second(i64),
+    /// TimeTriger in minute(s).
+    Minute(i64),
+    /// TimeTriger in hour(s).
+    Hour(i64),
+    /// TimeTriger in day(s).
+    Day(i64),
+    /// TimeTriger in week(s).
+    Week(i64),
+    /// TimeTriger in month(s).
+    Month(i64),
+    /// TimeTriger in year(s).
+    Year(i64),
+}
+
+impl Default for TimeTriggerInterval {
+    fn default() -> Self {
+        TimeTriggerInterval::Second(1)
+    }
+}
+
+#[cfg(feature = "config_parsing")]
+impl<'de> serde::Deserialize<'de> for TimeTriggerInterval {
```

Comment:
Now that we have the `modulate` field, we can use the `humantime` crate to parse this. You would then be able to set the roll time based either `current time + interval` or `start of day + interval` right?

Reply:
Right, so if I enter 4 hours as an interval, humantime will parse that as 14,400 seconds. The trigger would then be set like so:
- `modulate = true`: [start of date or last trigger time] + 14,400 seconds
- `modulate = false`: [startup time or last trigger time] + 14,400 seconds

I get the concern with humantime accounting for leap years, but as we are currently in a leap year, it does make sense to utilize them.

Reply:
I think we could reduce the code maintenance on this project by using humantime

Reply:
Understood and agreed on human time. With respect to the leap years, it's tough. Because with your implementation, it wouldn't work the current year we are in 2024. Over 4 years, the leap year makes sense. But in reality, I wouldn't expect anyone to maintain log files that long. I guess I'm okay with the current approach, and users can submit an issue if they feel strongly otherwise.